src/Decimal128.mts

```diff
@@ -166,10 +168,18 @@ export class Decimal128 {
     private readonly _isFinite: boolean = true;
     private readonly _isNegative: boolean = false;
 
-    constructor(n: string | number | bigint | Decimal) {
+    constructor(n: string | number | bigint | JSBI | Decimal) {
         let data;
         if ("object" === typeof n) {
-            data = n;
+            if (n instanceof JSBI) {
+                // Note: when using babel-plugin-transform-jsbi-to-bigint,
+                // the condition above may get transpiled into "typeof n === 'bigint'",
+                // causing the condition above to always be false.
+                // This is fine as we will be hanlding bigint below.
+                data = handleDecimalNotation(n.toString())
+            } else {
+                data = n;
+            }
```

Comment:
Note to reviewer:

Support type checking for both `bigint` and `JSBI`. Added disclaimer for `babel-plugin-transform-jsbi-to-bigint`